### PR TITLE
Sprint2 task update delete

### PR DESCRIPTION
**Pull request by ashinagawa**
Implemented "Edit Task" and "Remove Task" functionality in TaskCatalogView -> TaskDetailView flow:

- You can remove a Task in TaskCatalogView and it will update data in Parse Server
- You can Update (partially) a Task Details in "TaskDetailView" and it will update data in Parse Server

TODO:
- Make Task Details editable
- Fix the issue: Step data should be updated after edit in Step Cell
- Update TaskCatalogView after edit or remove
- Update HomeView after start a task and go to HomeView

walkthru_sprint2_edit_remove_task

### CODE REVIEW

File: Podfile.lock
Comment (sahilda) [inline Podfile.lock:38]: Y'all need to upgrade!

File: TaskHero/ParseClient.swift
Comment (sahilda) [inline TaskHero/ParseClient.swift:64]: ha nice catch

File: TaskHero/ParseClient.swift
Comment (sahilda) [inline TaskHero/ParseClient.swift:128]: probably not necessary to have this, but no harm either

File: TaskHero/ParseClient.swift
Comment (sahilda) [inline TaskHero/ParseClient.swift:147]: same case as above

File: TaskHero/ParseClient.swift
Comment (sahilda) [inline TaskHero/ParseClient.swift]: No need for state? I imagine we do.

File: TaskHero/TaskCardCell.swift
Comment (sahilda) [inline TaskHero/TaskCardCell.swift]: Look at the TaskInstance.getInvolvedUsers() method. You might want to implement the same thing into the Task model too, which you could then use here.

Comment (sahilda) [reply TaskHero/TaskCardCell.swift]: Better yet, maybe we could abstract this method into a TaskHelper class or something that takes both either a taskInstance or task as a parameter (not sure how this is done), so that it can be used for both cases.

File: Podfile.lock
Comment (akishinagawa) [reply Podfile.lock:38]: Updated.

File: TaskHero/ParseClient.swift
Comment (akishinagawa) [reply TaskHero/ParseClient.swift:128]: ok.

File: TaskHero/ParseClient.swift
Comment (akishinagawa) [reply TaskHero/ParseClient.swift:147]: we need to be able to update, don't we?

File: TaskHero/ParseClient.swift
Comment (akishinagawa) [reply TaskHero/ParseClient.swift]: I added them.

File: TaskHero/ParseClient.swift
Comment (sahilda) [inline TaskHero/ParseClient.swift]: Oh, sorry, I meant only this line: "query.order(byAscending: "createdAt")" might not be necessary since you're pulling a single object. This method is useful!

Sorry for the confusion :)

File: TaskHero/ParseClient.swift
Comment (sahilda) [inline TaskHero/ParseClient.swift]: Likewise, this line might be extra: query.order(byAscending: "createdAt"). Again, no harm, just code reviewing.

File: TaskHero/TaskCardCell.swift
Comment (akishinagawa) [reply TaskHero/TaskCardCell.swift]: I added "task.getInvolvedUsers" in Task class. But When I load all tasks (getAllTasks in ParseClient), somehow, assignees gets nil... Can you take a look at it?

For "TaskHelper" class,  "TaskInstance" needs to be  subclass of "Task Class"?

**Review by sahilda — APPROVED** [review]
Looks good to me. I can take a look at the assignees being nil after merge. @jcomo, I'll merge tomorrow if you want to take a look.

Never mind, talked with @aki about the problem - it's related to the Step model getting the users from parse in the background bug biting us. He's going to try to work on it and push changes, so going to hold off on merging.

File: TaskHero/ParseClient.swift
Comment (sahilda) [reply TaskHero/ParseClient.swift:128]: Oh, sorry, I meant only this line: "query.order(byAscending: "createdAt")" might not be necessary since you're pulling a single object. This method is useful!

Sorry for the confusion :)

File: TaskHero/ParseClient.swift
Comment (sahilda) [reply TaskHero/ParseClient.swift:147]: Again, just referencing the query.order(byAscending: "createdAt") line, but no biggie.

**Review by sahilda — APPROVED** [review]

**Pull request merged**
